pandas-stubs/__init__.pyi

```diff
@@ -39,7 +39,10 @@ def read_csv(
     squeeze: bool = ...,
     prefix: Optional[str] = ...,
     mangle_dupe_cols: bool = ...,
-    dtype: Optional[Union[Type, str, Dict[str, Union[Type, str]]]] = ...,
+    # Having Dict[str, Union[Type, str]] instead of
+    # Union[Dict[str, Type], Dict[str, str]] breaks the "Liskov substitution
+    # principle" and will result in mypy complaining when passing in a dicts.
+    dtype: Optional[Union[Type, str, Dict[str, Type], Dict[str, str]]] = ...,
```

Comment:
Would using `Mapping` instead of `Dict` help? (`Mapping` is covariant in the value.) I think some people legitimately pass things like `{"a": str, "b": "float32"}` to this function.

Reply:
Yes. `Mapping` helped.